python/dgl/graphbolt/itemset.py

```diff
@@ -178,7 +178,7 @@ def __getitem__(self, index: Union[int, slice, Iterable[int]]):
                     )
                 return torch.tensor(index, dtype=dtype)
             elif isinstance(index, Iterable):
-                return torch.tensor(index, dtype=dtype)
+                return index.type(dtype).clone().detach()
```

Comment:
1. if index is a PyTorch tensor, could we return `index.to(dtype)`?
2. Ideally we need change `elif isinstance(index, Iterable):` to `elif isinstance(index, torch.Tensor):`

Reply:
Done